### PR TITLE
Add integration tests for updating folder shares

### PR DESCRIPTION
**Pull request by juliusknorr**
As discussed with @rullzer

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Awesome thnx!

**Review by ChristophWurst — APPROVED** [review]
:rocket:

File: build/integration/features/sharing-v1-part3.feature [hunk 1]
```diff
@@ -393,3 +393,35 @@ Feature: sharing
...
+      | expireDate | +3 days |
+    And restore the last share data from "link"
+    And Getting info of last share
+    And Share fields of last share match with
+      | id | A_NUMBER |
+      | item_source | A_NUMBER |
+      | share_type | 3 |
+      | permissions | 23 |
+      | file_target | /subfolder |
+      | expiration  |            |
```
Comment (MorrisJobke) [inline build/integration/features/sharing-v1-part3.feature]:
```
Scenario: Link shares inside of group shares keep their original data when the root share is updated # /drone/src/build/integration/features/sharing-v1-part3.feature:397
  Given As an "admin" # FeatureContext::asAn()
  And user "user0" exists # FeatureContext::assureUserExists()
  And user "user1" exists # FeatureContext::assureUserExists()
  And group "group1" exists # FeatureContext::assureGroupExists()
  And user "user1" belongs to group "group1" # FeatureContext::assureUserBelongsToGroup()
  And As an "user0" # FeatureContext::asAn()
  And user "user0" created a folder "/share" # FeatureContext::userCreatedAFolder()
  And folder "/share" of user "user0" is shared with group "group1" # FeatureContext::assureFileIsSharedWithGroup()
  And user "user0" created a folder "/share/subfolder" # FeatureContext::userCreatedAFolder()
  And As an "user1" # FeatureContext::asAn()
  And save the last share data as "original" # FeatureContext::saveLastShareData()
  And as "user1" creating a share with # FeatureContext::asCreatingAShareWith()
    | path | /share/subfolder |
    | shareType | 3 |
    | permissions | 31 |
  And save the last share data as "link" # FeatureContext::saveLastShareData()
  And As an "user0" # FeatureContext::asAn()
  And restore the last share data from "original" # FeatureContext::restoreLastShareData()
  When Updating last share with # FeatureContext::updatingLastShare()
    | permissions | 23 |
    | expireDate | +3 days |
  And restore the last share data from "link" # FeatureContext::restoreLastShareData()
  And Getting info of last share # FeatureContext::gettingInfoOfLastShare()
  And Share fields of last share match with # FeatureContext::checkShareFields()
    | id | A_NUMBER |
    | item_source | A_NUMBER |
    | share_type | 3 |
    | permissions | 23 |
    | file_target | /subfolder |
    | expiration | |
    expiration doesn't have value
```

Comment (juliushaertl) [reply build/integration/features/sharing-v1-part3.feature]: Ah sorry, forgot to commit my squashed change after rebasing. Should be fine now.

**Pull request merged**